backend/src/main/java/com/wootech/dropthecode/domain/review/Review.java

```diff
@@ -63,9 +61,21 @@ public Review(Member teacher, Member student, String title, String content, Stri
         this.progress = progress;
     }
 
+    private void validateAuthorityOfReview(Member teacher, Member student) {
+        // 자기 자신에게 요청을 보내서 student와 teacher의 id가 같은 경우 예외 처리
+        if (student.hasSameId(teacher.getId())) {
+            throw new ReviewException("자신에게는 리뷰를 요청할 수 없습니다.");
+        }
+
+        // 학생이 학생한테 요청 보내서 teacher라고 들어왔지만 teacher 권한이 없는 경우 예외 처리
+        if (!teacher.hasRole(Role.TEACHER)) {
+            throw new ReviewException("리뷰어 권한이 없는 사용자에게는 리뷰를 요청할 수 없습니다.");
+        }
+    }
+
     public void validateAuthorityOfStudent(Long id) {
         if (!this.student.hasSameId(id)) {
-            throw new AuthorizationException("리뷰를 수정할 권한이 없습니다!");
+            throw new AuthorizationException("리뷰를 요청하거나 수정할 권한이 없습니다!");
```

Comment:
어떤 기능에 대한 예외인지 확실히 표시해주는 것은 좋지만, 예외 메시지를 조금 더 범용적으로 바꾸어서 예외 메시지를 수정하지 않도록 하는 것은 어떨까요?

만약 리뷰 삭제에도 권한이 필요하다면 위의 예외 메시지는

> 리뷰를 요청, 수정, 삭제에 대한 권한이 없습니다!

로 바뀔 것 같네요.

아래와 같이 CRUD를 크게 묶어서 메시지를 작성하면 좋을 것 같아요.

```suggestion
            throw new AuthorizationException("리뷰 작업에 대한 권한이 없습니다!");
```

`작업` 보다는 다른 좋은 단어로 하면 더 좋을 것 같네요.

Reply:
확실히 훨씬 범용적이네요😀